examples/publisher/README.md

```diff
@@ -0,0 +1,106 @@
+<!--
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2017 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+## Snap Plugin Library for C++: Publisher Plugin Example
+Here you will find an example plugin that covers the basics for writing a publisher plugin.
+
+## Plugin Naming, Files, and Directory
+For your publisher plugin, create a new repository and name your plugin project using the following format:
+
+>snap-plugin-[plugin-type]-[plugin-name]
+
+For example:
+>snap-plugin-publisher-log
+
+Proposed files and directory structure:  
+```
+snap-plugin-[plugin-type]-[plugin-name]
+ |--src
+  |--[plugin-name].cc  
+  |--[plugin-name]_test.cc  
+```
+
+For example:
+```
+snap-plugin-publisher-log
+ |--src
+  |--log.cc  
+  |--log_test.cc  
+```
+
+## Interface methods
+
+In order to write a plugin for Snap, it is necessary to define a few methods to satisfy the appropriate interfaces. These interfaces must be defined for a publisher plugin:
+
+```cpp
+/**
+ * The interface for a Publisher plugin.
+ * A Publisher is the sink.
+ * It sinks data into some external system.
+ */
+class PublisherInterface : public PluginInterface {
+public:
+  Type GetType() const final;
+  PublisherInterface* IsPublisher() final;
+
+  /*
+   * (inherited from PluginInterface)
+   */
+  virtual const ConfigPolicy get_config_policy() = 0;
+
+  virtual void publish_metrics(std::vector<Metric> &metrics,
+                               const Config& config) = 0;
+};
+
+```
+
+The interface is slightly different depending on what type (collector, processor, or publisher) of plugin is being written. Please see other plugin types for more details.
+
+## Starting a plugin
+
+After implementing a type that satisfies one of {collector, processor, publisher} interfaces, all that is left to do is to call the appropriate plugin.start_xxx() with your plugin specific meta options. For example with no meta options specified:
+
+```cpp
+    Plugin::start_publisher(&plg, Meta{Type::Publisher, "log", 1});
```

Comment:
There is mentioned an "example with no meta options specified", but this example has meta options. @marcintao , could you verify that and make it consistent?